### PR TITLE
feat: Added task-based historical allocation endpoint [DET-8537]

### PR DESCRIPTION
**Pull request by corban-beaird**
## Description
Previously the Historic Allocations CSV output was solely at a workload-granularity; however, it produced many rows resulting in the CSV being difficult to read. For [DET-8537], a new endpoint was added to create a CSV at a task-granularity. As well, a select was added to the `ClusterHistoricalUsageCsvModal` where the user can select between workload/task level output.

## Test Plan
e2e tests checking if the new task-level endpoint captures results from:
- No_Op Experiment
- Jupyter Notebook
- No_Op Experiment + Tensorboard

Validating that the endpoint returned successfully and that the CSV result returned an entry for each case

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket
DET-8537

[DET-8537]: https://determinedai.atlassian.net/browse/DET-8537?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: e2e_tests/tests/experiment/test_tasks_raw_csv.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+from tests import experiment as exp
+
+
+# Create a No_Op experiment
+#  and confirm training and validation times are captured
+@pytest.mark.e2e_cpu
+def test_experiment_capture() -> None:
+    start_time = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    # Create Experiment & wait for completion
```
[the remark this answers is not in the document]
Comment (corban-beaird) [reply e2e_tests/tests/experiment/test_tasks_raw_csv.py:21]: No worries! I tend to err on the over-explaining side of things. I'll strip out the W.E.T comments

File: master/internal/rm/kubernetesrm/request_queue.go [hunk 2]
```diff
@@ -59,41 +59,44 @@ type queuedResourceRequest struct {
...
-//
-//  The message protocol consists of `createKubernetesResources` and `deleteKubernetesResources`
-//  messages being sent to the requestQueue. If it forwards the request to a worker, the worker
-//  will send the original task handler a `resourceCreationFailed` or a `resourceDeletionFailed`
-//  if an error was encountered while creating / deleting the resources. If a deletion request
-//  arrives before the creation request had been sent to the worker, the `requestQueue` will
-//  notify the task handler of this by sending a `resourceCreationCancelled` message.
-//  requestProcessingWorkers notify the requestQueue that they are available to receive work
-//  by sending a `workerAvailable` message.
+/*
```
[the remark this answers is not in the document]
Comment (corban-beaird) [reply master/internal/rm/kubernetesrm/request_queue.go]: Got this resolved, the linter had been giving me some issues due to Go version differences (should have been using 1.18, but was using 1.19)

File: e2e_tests/tests/experiment/test_tasks_raw_csv.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        f"/resources/allocation/tasks-raw?timestamp_after={start_time}&timestamp_before={end_time}",
+    )
+    assert r.status_code == requests.codes.ok, r.text
+
+    # Check if a trial entry exists for experiment that just ran
+    matches = re.search(f".+,TRIAL,.+,{experiment_id},.+", r.text)
+    assert matches is not None
+
+    # Confirm valid training and validation time were captured for the given experiment
+    training_time, validation_time = matches.group(0).split(",")[-3:-1]
```
Comment (ioga) [inline e2e_tests/tests/experiment/test_tasks_raw_csv.py:37]: maybe parse it in a more stable way, as a CSV, so potential column changes don't affect it?

```suggestion
    reader = csv.DictReader(StringIO(r.text))
    matches = [row for row in reader if int(row["experiment_id"]) == experiment_id]
    assert len(matches) == 1
    training_time, validation_time = matches[0]["training_time"], matches[0]["validation_time"]
```

File: master/internal/core.go [hunk 4]
```diff
@@ -314,12 +315,515 @@ func (m *Master) fetchAggregatedResourceAllocation(
...
+//	@Param		timestamp_before	query	string	true	"End time to get allocations for (YYYY-MM-DDTHH:MM:SSZ format)"
+// nolint:lll
+//	@Success	200					{}		string	"A CSV file containing the fields task_id, task_type, username, workspace_name, experiment_id, slots, start_time, end_time, training_time, validation_time, checkpointing_time, imagepulling_time"
+//	@Router		/allocations/tasks-raw [get]
+func (m *Master) getRawResourceAllocationTasks(c echo.Context) error {
+	// Get start and end times from context
+	args := struct {
+		Start string `query:"timestamp_after"`
+		End   string `query:"timestamp_before"`
+	}{}
```
Comment (ioga) [inline master/internal/core.go:602]: hmmm is this a duplicate of the code above, rebase issue?

File: e2e_tests/tests/experiment/test_tasks_raw_csv.py (hunk 3 above)
Comment (corban-beaird) [reply e2e_tests/tests/experiment/test_tasks_raw_csv.py]: Makes a lot of sense! Committing that change

**Review by nrajanee — APPROVED** [review]
lgtm

**Pull request merged**